cluster-controller/src/main/java/io/strimzi/controller/cluster/operations/cluster/KafkaClusterOperations.java

```diff
@@ -451,13 +543,74 @@ protected void delete(String namespace, String name, Handler<AsyncResult<Void>>
+
+        @Override
+        public Future<?> composite(String namespace, ClusterOperation<TopicController> operation) {
+            TopicController topicController = operation.cluster();
+            ClusterDiffResult diff = operation.diff();
+
+            Future<Void> fut;
+            if (diff != null) {
+                fut = patchDeployment(topicController, namespace, diff);
+            } else {
+                // that's the case we are moving from no topic controller deployed to an updated cluster ConfigMap
+                // which contains topic controller configuration for deploying it (so there is no Deployment to patch
+                // but to create)
+                fut = deploymentOperations.create(topicController.generateDeployment());
+            }
+
+            return fut;
+        }
+
+        @Override
+        public ClusterOperation<TopicController> getCluster(String namespace, String name) {
+            ClusterDiffResult diff = null;
+            TopicController topicController = null;
+            ConfigMap tcConfigMap = configMapOperations.get(namespace, name);
+
+            if (tcConfigMap != null) {
+                topicController = TopicController.fromConfigMap(tcConfigMap);
+                if (topicController != null) {
+                    log.info("Updating Topic Controller {} in namespace {}", topicController.getName(), namespace);
+                    Deployment dep = deploymentOperations.get(namespace, topicController.getName());
+                    diff = topicController.diff(dep);
+                }
+            } else {
+                throw new IllegalStateException("ConfigMap " + name + " doesn't exist anymore in namespace " + namespace);
+            }
+
+            return new ClusterOperation<>(topicController, diff);
+        }
+
+        private Future<Void> patchDeployment(TopicController topicController, String namespace, ClusterDiffResult diff) {
+            if (diff.isDifferent()) {
+                return deploymentOperations.patch(namespace, topicController.getName(),
+                        topicController.patchDeployment(deploymentOperations.get(namespace, topicController.getName())));
+            } else {
+                return Future.succeededFuture();
+            }
+        }
+    };
+
     @Override
     public void update(String namespace, String name, Handler<AsyncResult<Void>> handler) {
-        execute(CLUSTER_TYPE_ZOOKEEPER, OP_UPDATE, namespace, name, updateZk, ar -> {
-            if (ar.failed()) {
-                handler.handle(ar);
+        execute(CLUSTER_TYPE_ZOOKEEPER, OP_UPDATE, namespace, name, updateZk, zookeeperDone -> {
+            if (zookeeperDone.failed()) {
+                handler.handle(zookeeperDone);
             } else {
-                execute(CLUSTER_TYPE_KAFKA, OP_UPDATE, namespace, name, updateKafka, handler);
+                execute(CLUSTER_TYPE_KAFKA, OP_UPDATE, namespace, name, updateKafka, kafkaDone -> {
```

Comment:
These `*Done` for the handlers... elsewhere I've used `*Result` and still other places I've just used `ar`. We need to converge on one style, but adding a 3rd seems like a step backward.

Reply:
What's the style do you prefer ?

Reply:
Not `ar`, I suppose, even though they're probably all mine. I could live with either of the other two, but

```
grep -lr '[a-zA-Z0-9]Result' . | wc -l
167
```

suggests `*Result` would be much easier at this point

Reply:
Done